src/util.h

```diff
@@ -125,16 +126,23 @@ void DumpJavaScriptBacktrace(FILE* fp);
 #define ABORT_NO_BACKTRACE() abort()
 #endif
 
-#define ABORT() node::Abort()
+#define ABORT()                                                                \
+  do {                                                                         \
+    node::DumpNativeBacktrace(stderr);                                         \
+    node::DumpJavaScriptBacktrace(stderr);                                     \
+    fflush(stderr);                                                            \
+    ABORT_NO_BACKTRACE();                                                      \
+  } while (0)
 
-#define ERROR_AND_ABORT(expr)                                                 \
-  do {                                                                        \
-    /* Make sure that this struct does not end up in inline code, but      */ \
-    /* rather in a read-only data section when modifying this code.        */ \
-    static const node::AssertionInfo args = {                                 \
-      __FILE__ ":" STRINGIFY(__LINE__), #expr, PRETTY_FUNCTION_NAME           \
-    };                                                                        \
-    node::Assert(args);                                                       \
+#define ERROR_AND_ABORT(expr)                                                  \
+  do {                                                                         \
+    /* Make sure that this struct does not end up in inline code, but      */  \
+    /* rather in a read-only data section when modifying this code.        */  \
+    static const node::AssertionInfo args = {                                  \
+        __FILE__ ":" STRINGIFY(__LINE__), #expr, PRETTY_FUNCTION_NAME};        \
+    node::Assert(args);                                                        \
+    /* Mark the macro as [[noreturn]].                                     */  \
```

Comment:
I think it's not actually not `[[noreturn]]` on all platforms, and then we don't care much about `[[noreturn]]` here because we are not dumping backtraces from `ABORT_NO_BACKTRACE` anyway...we could just add a comment to `ABORT()` and `ERROR_AND_ABORT()` to remind people not to mark any caller of them as `[[noretun]]`.

Reply:
By this comment, I mean that this macro would not yield back to the caller, the same as calling a `[[noreturn]]` function.

Reply:
I was confused because I don't think there's such thing as "marking a macro as [[noreturn]]" - macros don't even return. Maybe this can just be replaced by a comment above the macro reminding callers not to mark themselves as `[[noreturn]]`?

Reply:
Updated as explained at https://github.com/nodejs/node/pull/50849#discussion_r1407777419.